nixos/modules/services/databases/mysql.nix

```diff
@@ -273,6 +273,9 @@ in
       datadir = ${cfg.dataDir}
       ${optionalString (cfg.bind != null) "bind-address = ${cfg.bind}" }
       ${optionalString (cfg.replication.role == "master" || cfg.replication.role == "slave") "log-bin=mysql-bin"}
+      ${optionalString (cfg.replication.role == "master" || cfg.replication.role == "slave") "log-bin=mysql-bin-${toString cfg.replication.serverId}"}
```

Comment:
also, these conditions probably can be simplified, as they're all the same.

Reply:
Thanks, removed duplicate

Reply:
Please add a single if-statement checking `(cfg.replication.role == "master" || cfg.replication.role == "slave")`, with all options underneath.

Reply:
Fixed